### PR TITLE
[CSLD-151] Rewrite PoS chapter.

### PR DESCRIPTION
**Pull request by denisshevchenko**

### CODE REVIEW

File: _docs/cardano/2017-01-01-proof-of-stake.md [hunk 1]
```diff
@@ -9,83 +9,173 @@ visible: true
...
 
 ## What is Proof of Stake?
 
-In this section we explain what “proof” and “stake” mean, and then put it all
-together, explaining what “proof of stake” means.
-
 ### Proof
 
-The “proof” part of "proof of stake" is about having evidence that blocks of
+The "proof" part of "proof of stake" is about having evidence that blocks of
```
Comment (gromakovsky) [inline _docs/cardano/2017-01-01-proof-of-stake.md]: Why did you change `“”` to `""`?

File: _docs/cardano/2017-01-01-proof-of-stake.md [hunk 2]
```diff
@@ -9,83 +9,173 @@ visible: true
...
+  | Bl 0 |<--| Bl 1 |<-- ... <--| Bl N |
+  +------+   +------+           +------+
++----------+----------+-------+----------+----> t
+|  slot 0  |  slot 1  |  ...  |  slot N  |
+
+    SL 0       SL 1               SL N
+```
+
+It means that the number of slot leaders is strictly equal to the number of slots
+in epoch (let's call this number `N`), so it is impossible to produce more than `N`
```
Comment (gromakovsky) [inline _docs/cardano/2017-01-01-proof-of-stake.md:93]: It's true if slot leaders are represented as a list and not necessary true if they are represented as a set. I don't know whether reader will understand it correctly 🤷‍♂️

File: _docs/cardano/2017-01-01-proof-of-stake.md [hunk 3]
```diff
@@ -9,83 +9,173 @@ visible: true
...
+
+If slot leader missed his slot (for example, he was offline at that moment), he
+lost his right to produce a block until he will be elected again. So one or more
+slots can remain empty (i.e. without generated blocks), but please note that the
+most of blocks (i.e. at least 50% + 1) **must** be generated during an epoch.
+
+## Slot Leaders Election
+
+Slot leaders are elected from among stakeholders. Please note that not all stakeholders
+participate in this election, but only ones who has enough stake (for example, 2% of
```
Comment (gromakovsky) [inline _docs/cardano/2017-01-01-proof-of-stake.md]: I guess it can be interpreted incorrectly by potential reader. One possible interpretation is that only stakeholders with at least 2% stake can become slot leaders. But it's not true, even someone with 1 ADA can become a slot leader. However, only stakeholders with at least 2% stake can affect leader selection process.

Comment (gromakovsky) [reply _docs/cardano/2017-01-01-proof-of-stake.md]: Upd: I see that later you say that everyone can be a slot leader, but I still think this sentence may lead to some confusion. It's subjective though, maybe it's just me who thinks so.

File: _docs/cardano/2017-01-01-proof-of-stake.md [hunk 4]
```diff
@@ -9,83 +9,173 @@ visible: true
...
+#### Commitment Phase
+
+First of all, elector generates a secret (special random value). Then elector forms a
+"commitment" which is a message that contains encrypted shares and proof of secret.
+
+Then elector signs this commitment with its secret key, specifies epoch's number and attaches
+its public key. In this case everybody can check who created this commitment and which epoch
+this commitment is for.
+
+After that elector send its commitment to other electors, so eventually each elector collects
```
Comment (gromakovsky) [inline _docs/cardano/2017-01-01-proof-of-stake.md]: s/send/sends

File: _docs/cardano/2017-01-01-proof-of-stake.md [hunk 5]
```diff
@@ -9,83 +9,173 @@ visible: true
...
+commitments from all other electors.
```
Comment (gromakovsky) [inline _docs/cardano/2017-01-01-proof-of-stake.md]: I'd add that commitments are put into blocks, they are part of blockchain.

File: _docs/cardano/2017-01-01-proof-of-stake.md [hunk 6]
```diff
@@ -9,83 +9,173 @@ visible: true
...
+#### Opening Phase
+
+Now elector sends an "opening", special value for opening a commitment. You can think of a
+commitment as a locked box (with a secret in it), and opening is a key we need to open this
+box and get the secret from it.
+
+So eventually elector has commitments and openings. Then elector verifies that commitments
+and openings are match, and if so, he gets the secrets from commitment and forms a seed from
+these secrets. So all electors get the same seed, and it will be used for Follow the Satoshi
+algorithm.
```
Comment (gromakovsky) [inline _docs/cardano/2017-01-01-proof-of-stake.md]: What about Shares Phase? Is it intentionally omitted for simplicity?

File: _docs/cardano/2017-01-01-proof-of-stake.md [hunk 7]
```diff
@@ -9,83 +9,173 @@ visible: true
...
+
+Let's elaborate a little bit on how a slot leader gets selected. The smallest, atomic piece
+of value is a coin, we call it "[Lovelace](/glossary/#lovelace)". Fundamentally, we can say
+that the ledger produces distribution of coins, and since slot leaders can be selected from
+stakeholders only, we are talking about distribution of stake. FTS is an algorithm that
+verifiably picks a coin, and when coin owned by stakeholder `S` gets selected, `S` become a
+slot leader. It is obviously that the more coins `S` has, the higher probability that one of
+his coins will be picked.
+
+The reason why does it called "Follow the Satoshi" is that in Bitcoin, atomic piece of currency
```
Comment (gromakovsky) [inline _docs/cardano/2017-01-01-proof-of-stake.md]: The reason why it is called…

File: _docs/cardano/2017-01-01-proof-of-stake.md [hunk 8]
```diff
@@ -9,83 +9,173 @@ visible: true
...
+is called "Satoshi", honoring Satoshi Nakamoto, the creator of Bitcoin. 
+
+## Honest Majority
+
+The fundamental assumption of a protocol is a **honest majority**. It means that
+the most of protocol participants (i.e. at least 50% + 1) are honest ones. In
```
Comment (gromakovsky) [inline _docs/cardano/2017-01-01-proof-of-stake.md]: Not `50% + 1` participants (participants are nodes), but participants owning `50% + 1` of stake. It can be only two participants each holding 30% of coins.

File: _docs/cardano/2017-01-01-proof-of-stake.md (hunk 6 above)
Comment (denisshevchenko) [reply _docs/cardano/2017-01-01-proof-of-stake.md]: Yes, for simplicity. Do you think I must add it?

Comment (gromakovsky) [reply _docs/cardano/2017-01-01-proof-of-stake.md]: I think it'd be good to add it, at least one or two sentences, because otherwise someone who will also read Ouroboros paper might be confused. There is no need to describe details, just write that Shares Phase exists to prevent possible confusion.

File: _docs/cardano/2017-01-01-proof-of-stake.md [hunk 9]
```diff
@@ -9,83 +9,174 @@ visible: true
...
+
+If slot leader missed his slot (for example, he was offline at that moment), he
+lost his right to produce a block until he will be elected again. So one or more
+slots can remain empty (i.e. without generated blocks), but please note that the
+most of blocks (i.e. at least 50% + 1) **must** be generated during an epoch.
+
+## Slot Leaders Election
+
+Slot leaders are elected from among all stakeholders. Please note that not all stakeholders
+participate in this election, but only ones who has enough stake (for example, 2% of
```
Comment (gromakovsky) [inline _docs/cardano/2017-01-01-proof-of-stake.md]: s/has/have I guess

File: _docs/cardano/2017-01-01-proof-of-stake.md [hunk 10]
```diff
@@ -9,83 +9,174 @@ visible: true
...
+blocks during an epoch.
+
+If slot leader missed his slot (for example, he was offline at that moment), he
+lost his right to produce a block until he will be elected again. So one or more
+slots can remain empty (i.e. without generated blocks), but please note that the
+most of blocks (i.e. at least 50% + 1) **must** be generated during an epoch.
+
+## Slot Leaders Election
+
+Slot leaders are elected from among all stakeholders. Please note that not all stakeholders
```
Comment (gromakovsky) [inline _docs/cardano/2017-01-01-proof-of-stake.md:103]: `all` indeed makes it more clear :)

**Review by gromakovsky — APPROVED** [review]
Looks good now. I'd add few words about Shares Phase (see my comment), but it's optional and I can approve even without that (will review this part if you add it).

File: _docs/cardano/2017-01-01-proof-of-stake.md [hunk 11]
```diff
@@ -9,83 +9,187 @@ visible: true
...
+#### Recovery Phase
+
+This is a final phase.
+
+Eventually elector has commitments and openings. But theoretically some elector can be an
+adversary. And he could publish its commitment but **not** publish its opening.
+
+In this case the honest electors can post all shares (mentioned above) to reconstruct the
+secret. The idea is simple: we have to be sure that election will finish successfully even
+if some of electors is an adversary.
```
Comment (gromakovsky) [inline _docs/cardano/2017-01-01-proof-of-stake.md]: are adversary (electors is plural)
